src/Crawler.php

```diff
@@ -371,13 +371,14 @@ protected function addAllLinksToCrawlQueue(string $html, UriInterface $foundOnUr
                 return $this->crawlQueue->has($url);
             })
             ->each(function (UriInterface $url) use ($foundOnUrl) {
-                $node = $this->addtoDepthTree($this->depthTree, $url, $foundOnUrl);
-
-                if (strpos($url->getPath(), '/tel:') === 0) {
-                    return;
+                if ($this->maximumDepth !== null) {
```

Comment:
Change this to 

```
if (is_null($this-> maximumDepth) {
   return;
}
```

so we use an indentation level in the rest of the function.

Reply:
with that change all $url would return; because maximunDepth never changes.
I've anohter idea, changing shoulCrawl signature to
```php
/**
* @param Node|null $node
*/
protected function shouldCrawl( $node): bool
```
and placing at the begining of addtoDepthTree a is_null check
```php
protected function addtoDepthTree(Node $node, UriInterface $url, UriInterface $parentUrl)
    {
        $returnNode = null;
if (is_null($this-> maximumDepth)) {
return $returnNode;
}
```

Reply:
Another idea, changing shouldCrawl signature so it can call addtoDepthTree

Reply:
> with that change all $url would return; because maximunDepth never changes.
I don't understand your argument.

Reply:
Why prefer `is_null` instead of `=== null` ? It prevents function call and with a lots of call, this can reduce function call overhead, no ?

Reply:
Avoiding `is_null` is a micro optimization. We use `is_null` in other packages so i'd like to keep it consistent.

My main gripe was just reversing the condition so we can lose a level of indentation.

Reply:
Just making that change would have prevented to add any url to the crawl queue. Instead I moved the the check inside addToDepthTree and now returns a Node without checking or adding it to the depth tree.